Percolation.java

```diff
@@ -3,12 +3,79 @@
 import edu.princeton.cs.algs4.WeightedQuickUnionUF;
 
 public class Percolation {
-   public Percolation(int n)                // create n-by-n grid, with all sites blocked
-   public    void open(int row, int col)    // open site (row, col) if it is not open already
+   public boolean[][] grid;
+   private int nos  = N;
+   public WeightedQuickUnionUF connections;
+   private StdRandom randomGenerator;
+   
+   public Percolation(int n) {                // create n-by-n grid, with all sites blocked
+        nos = N;
+        boolean[][] grid = new boolean[N][N];
+        connect = new WeightedQuickUnionUF(N * N);
+        al = new ArrayList<Integer>();
+        randomGenerator = new Random();
+        
+        for (int i = 0; i < N; i++) {
+            for (int j = 0; j < N; j++) {
+                grid[i][j] = false;
+            }
+        }
+        
+   }
+   
+   private int toNum(int row, int col){
+       int num = col + ((row-1) * N) -1;
+       return num;
+   }    
+   public void open(int row, int col) {    // open site (row, col) if it is not open already
+       grid[i - 1][j - 1] = true;         // (i - 1, j - 1) is the site I just opened
```

Comment:
What is i and j? Where does it come from? sorry I can't see how it gets passed to this method...